app/controllers/sks_users_controller.ts

```diff
@@ -0,0 +1,123 @@
+import { HttpContext } from '@adonisjs/core/http'
+import SksUser from '#models/sks_user'
+import { DateTime } from 'luxon'
+
+//this value determines the period over which the trend will be counted (multiply the value by 5 minutes)
+const trendDelta = 3
+
+enum Trend {
+  INCREASING = 'INCREASING',
+  DECREASING = 'DECREASING',
+  STABLE = 'STABLE',
+}
+
+export default class SksUsersController {
+  /**
+   * Display the latest SKS users record based on time
+   */
+  async latest({ response }: HttpContext) {
+    try {
+      const currentTime = DateTime.now().setZone('Europe/Warsaw').toSQL()
+      if (currentTime === null) {
+        return response.status(500).json({ message: 'Failed to convert time to SQL format' })
+      }
+
+      const latestData = await SksUser.query()
+        .where('externalTimestamp', '<', currentTime)
+        .orderBy('externalTimestamp', 'desc')
+        .first()
+
+      const isResultRecent = latestData !== null && latestData.activeUsers > 0
+
+      // If the first record has activeUsers set to 0, get the second record instead
+      const entryToReturn = isResultRecent
+        ? latestData
+        : await SksUser.query()
+            .where('externalTimestamp', '<', currentTime)
+            .orderBy('externalTimestamp', 'desc')
+            .offset(1)
+            .first()
+
+      if (entryToReturn === null) {
+        return response
+          .status(404)
+          .json({ message: 'Could not find the matching data in database' })
+      }
+
+      const referenceTime = entryToReturn.externalTimestamp
```

Comment:
<img width="509" alt="image" src="https://github.com/user-attachments/assets/e6623b07-eee0-4bc3-873a-11235ad3f185">
there is some weird stuff going on with date time format (both externalTimestamp and created/updatedAt are of DateTime type but have different formats) and therefore i can't convert it to sql string

this version actually seems to look fine but i get type error